sycl/include/sycl/ext/oneapi/experimental/annotated_ptr/annotated_ptr.hpp

```diff
@@ -408,7 +408,16 @@ __SYCL_TYPE(annotated_ptr) annotated_ptr<T, detail::properties_t<Props...>> {
 
   operator T *() const noexcept = delete;
 
-  T *get() const noexcept { return m_Ptr; }
+  // T *get() const noexcept {
+  T *get() const noexcept {
+#ifdef __SYCL_DEVICE_ONLY__
+    return __builtin_intel_sycl_ptr_annotation(
+          m_Ptr, detail::PropertyMetaInfo<Props>::name...,
+          detail::PropertyMetaInfo<Props>::value...);
```

Comment:
This will annotate the pointer with _all_ its properties, right?  I don't think this is generally legal; this should be property dependent.  For example, if a property was worded like this then you would not be allowed to apply it in this case:

"foo: the foo property directs the compiler to ensure that all accesses through this annotated_ptr should be done using foo loads or stores."  

On the other hand, if it's worded like this (like alignment) then it is legal to apply it:

"foo: the foo property indicates that this pointer has foo characteristics."

The key difference is whether the property is information that the user is telling us about the pointer or whether it's a directive from the user for us to treat all dereferences of the pointer in a specific way.

Reply:
The code has been updated by adding a filter for the properties in `Props...`.